replication-api/src/main/java/org/codice/ditto/replication/api/modern/ReplicationPersistentStore.java

```diff
@@ -0,0 +1,55 @@
+/**
+ * Copyright (c) Connexta
+ *
+ * <p>This is free software: you can redistribute it and/or modify it under the terms of the GNU
+ * Lesser General Public License as published by the Free Software Foundation, either version 3 of
+ * the License, or any later version.
+ *
+ * <p>This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY;
+ * without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Lesser General Public License for more details. A copy of the GNU Lesser General Public
+ * License is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ditto.replication.api.modern;
+
+import java.util.List;
+import java.util.Optional;
+
+/** A ReplicationPersistentStore performs CRUD operations for replication. */
+public interface ReplicationPersistentStore {
+
+  /**
+   * Attempts to find an object that has been saved with the given persistence type and ID. It then
+   * creates an instance of class {@code <T>} with the data retrieved from persistence.
+   *
+   * @param klass The class of object to instantiate and return.
+   * @param typeName The persistence type of the desired object.
+   * @param id The id of the object to search for.
+   * @param <T> A class that implements the Persistable interface. An object of this type will be
+   *     created from the persisted data and returned.
+   * @return The object that was retrieved from storage
+   */
+  <T extends Persistable> Optional<T> get(Class<T> klass, String typeName, String id);
```

Comment:
❓ What if we are unable to retrieve the objects from storage?

Reply:
Agreed. As commented before, such distinction is required from a client's perspective

Reply:
❓ Why both a class and a type? Are those not linked by design? If a client specifies a class, don't we know for fact which type he/she wants? Plus it now requires the client to know all the types. Where do we define these? Inside the class?